### PR TITLE
Persistence added

### PR DESCRIPTION
**Pull request by axamon**
I added the possibily to persist a trained network on file and some examples

### CODE REVIEW

File: example/01/main.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/axamon/gobrain"
```
Comment (jonhkr) [inline example/01/main.go]: Shouldn't it be `github.com/goml/gobrain`?

File: example/01/main_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+   // "testing"
+	"math/rand"
+	"github.com/axamon/gobrain"
```
Comment (jonhkr) [inline example/01/main_test.go]: Shouldn't it be `github.com/goml/gobrain`?

File: example/02/main.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+
+	"github.com/axamon/gobrain"
```
Comment (jonhkr) [inline example/02/main.go]: Shouldn't it be `github.com/goml/gobrain`?

File: example/02/main_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+   // "testing"
+   "log"
+	"math/rand"
+	"github.com/axamon/gobrain"
```
Comment (jonhkr) [inline example/02/main_test.go]: Shouldn't it be `github.com/goml/gobrain`?

File: example/02/main_test.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    // create the XOR representation patter to train the network
+    patterns := [][][]float64{
+      {{0, 0}, {0}},
+      {{0, 1}, {1}},
+      {{1, 0}, {1}},
+      {{1, 1}, {0}},
+    }
+
+    // instantiate the Feed Forward
+	ff := &gobrain.FeedForward{}
```
Comment (jonhkr) [inline example/02/main_test.go]: There is something wrong with the indentation

File: persist/persist.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	b, err := json.MarshalIndent(v, "", "\t")
+	if err != nil {
+		return nil, err
+	}
+	return bytes.NewReader(b), nil
+}
+
+// Save saves a representation of v to the file at path.
+func Save(path string, v interface{}) error {
+	lock.Lock()
```
Comment (jonhkr) [inline persist/persist.go]: Why do we need a lock here?

File: example/01/main.go (hunk 1 above)
Comment (axamon) [reply example/01/main.go]: yea fixed it

File: example/01/main_test.go (hunk 2 above)
Comment (axamon) [reply example/01/main_test.go]: yea fixed it

File: example/02/main_test.go (hunk 4 above)
Comment (axamon) [reply example/02/main_test.go]: yea fixed it

File: example/02/main_test.go (hunk 5 above)
Comment (axamon) [reply example/02/main_test.go]: nop, don't see it

File: persist/persist.go (hunk 6 above)
Comment (axamon) [reply persist/persist.go]: to avoid having something else mess with it until saving is done

File: example/02/main_test.go (hunk 5 above)
Comment (axamon) [reply example/02/main_test.go]: I guess it was solved in some other commit

File: example/02/main.go (hunk 3 above)
Comment (axamon) [reply example/02/main.go]: yea, fixed it

File: persist/persist.go (hunk 6 above)
Comment (ebbot654) [reply persist/persist.go]: For it to have any effect the "something else" must also use the `Mutex`, right?

**Pull request merged**